docs/source/federation/migrating-from-stitching.md

```diff
@@ -103,11 +53,30 @@ const server = new ApolloServer({
+
+### Starting up a new Gateway
+
+Once you've migrated your first service to federation, you can start exposing them through a top-level gateway. Apollo gateway is a query-planner and executor that handles incoming GraphQL requests and breaks them down into a series of operations to underlying GraphQL services. We recommend setting the Apollo gateway up side-by-side with your existing stitching gateway. Depending on your infrastructure, you may even want to put them in the same process to support dynamically routing traffic through either your stitching gateway or the federated gateway.
+
+As mentioned before, we highly recommend running the gateway with a [managed configuration](https://www.apollographql.com/docs/graph-manager/federation/#managed-configuration). To create a gateway that uses managed configuration through Apollo Graph Manager, you simply need to set `ENGINE_API_KEY` and `ENGINE_SCHEMA_TAG` environment variables appropriately and leave out the `serviceList` constructor option to `ApolloGateway`. See the [Graph Manager docs](https://www.apollographql.com/docs/graph-manager/federation/#connecting-apollo-server-to-the-graph-manager) for more details.
 
-Next, you need to modify your existing schema to take advantage of federation. For more information on how to do this, check out [this guide](/federation/core-concepts/).
+Once your gateway is set up, you should be able to make direct queries to it that get routed to underlying services and see responses come through.
 
-### Removing type extensions from the gateway
+### Moving linking logic to service
+
+When running a stitching gateway, it's customary to put linking logic _in the gateway_ rather than in the services themselves. The federation model, on the other hand, puts linking logic in services, where resolvers are written to extend foreign types using a primary `@key` field on that type as a dependency. Thus, as part of the migration from schema-stitching to federation, the next step will be to migrate this logic from your stitching gateway into downstream services. Below, we've included some examples of how to do this in more detail. For the most common use cases, here is a good rule of thumb to follow:
+
+* **fragments** : When using fragments in your schema stitching resolvers, these usually translate to a combination of `@key` and `@requires` directives. In general, think of `@key` as the field(s) that completely identify an entity, and only use `@requires` for additional non-identifying information
+* **filtering types** : We do not recommend filtering types out of your exposed schema when using a gateway. If you want types to be hidden, simply do not include them in your service's registered SDL.
```

Comment:
> "simply do not include them in your service's registered SDL."

Should this have an example or further explanation?  Schema masking or other types of separation are a pattern for larger, more complex use cases.

Reply:
I think this is actually something that we need to support with an `@internal` directive tbh.